### PR TITLE
Add OrchardCoreConstants

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace OrchardCore;
+
+public class OrchardCoreConstants
+{
+    public const string TenantsFileName = "tenants.json";
+
+    public const string ApplicationSettingsFileName = "tenants.json";
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:7]:
```suggestion
    public const string ApplicationSettingsFileName = "appsettings.json";
```

File: src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace OrchardCore;
+
+public class OrchardCoreConstants
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: @hishamco I like what you did here but I have a suggestion.

If you want to call this file `OrchardCoreConstats` I would suggest that you sub section it to accommodate other contacts for example 

```csharp
public class OrchardCoreConstants
{
    public class Shell
    {
        public const string TenantFileName = "tenant.json";
    }


    public class Role
    {
        public const string Administrator = "Administrator";

    }
}
```


If you don't like that idea, then you should rename this file to `ShellConstants` instead.

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

File: src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs (hunk 2 above)
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: There's no role here, you might mean application settings

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: I meant role when we add role constants.

The name OrchardCoreConstants is very vague to use here unless you want to put all the commons constants here like suggested above

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: > The name OrchardCoreConstants is very vague

Right, but we usually name the constant class based on the project, your suggestion then we can finalize this PR

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: My suggestion is here https://github.com/OrchardCMS/OrchardCore/pull/13438#discussion_r1477125957

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs:3]: While I don't have a role in my PR, I used the `Shell` class also, added a `Configuration` one. Did I miss anything?

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**